ESP32-IDF: WPA Enterprise fails to compile

### The problem

Enabling WPA Enterprise causes compile failure.

### Which version of ESPHome has the issue?

2021.10.2

### What type of installation are you using?

Docker

### Which version of Home Assistant has the issue?

NA

### What platform are you using?

ESP32-IDF

### Board

esp32-c3-devkitm-1

### Component causing the issue

wifi

### Example YAML snippet

```yaml
---
esphome:
  name: bug

esp32:
  board: esp32-c3-devkitm-1
  framework:
    type: esp-idf

wifi:
    ssid: wpa-peap
    eap:
      username: foo
      password: bar
```


### Anything in the logs that might be useful for us?

```txt
src/esphome/components/wifi/wifi_component_esp_idf.cpp: In member function 'bool esphome::wifi::WiFiComponent::wifi_sta_connect_(const esphome::wifi::WiFiAP&)':
src/esphome/components/wifi/wifi_component_esp_idf.cpp:378:5: error: 'esp_wpa2_config_t' was not declared in this scope
     esp_wpa2_config_t wpa2_config = WPA2_CONFIG_INIT_DEFAULT();
     ^~~~~~~~~~~~~~~~~
src/esphome/components/wifi/wifi_component_esp_idf.cpp:378:5: note: suggested alternative: 'esp_eth_config_t'
     esp_wpa2_config_t wpa2_config = WPA2_CONFIG_INIT_DEFAULT();
     ^~~~~~~~~~~~~~~~~
     esp_eth_config_t
src/esphome/components/wifi/wifi_component_esp_idf.cpp:379:41: error: 'wpa2_config' was not declared in this scope
     err = esp_wifi_sta_wpa2_ent_enable(&wpa2_config);
                                         ^~~~~~~~~~~
src/esphome/components/wifi/wifi_component_esp_idf.cpp:379:41: note: suggested alternative: 'dump_config'
     err = esp_wifi_sta_wpa2_ent_enable(&wpa2_config);
                                         ^~~~~~~~~~~
                                         dump_config
Compiling .pioenvs/bug/asio/asio/asio/src/asio.o
*** [.pioenvs/bug/src/esphome/components/wifi/wifi_component_esp_idf.o] Error 1
```


### Additional information

_No response_

--- a/esphome/components/wifi/wifi_component_esp_idf.cpp
+++ b/esphome/components/wifi/wifi_component_esp_idf.cpp
@@ -375,8 +375,7 @@ bool WiFiComponent::wifi_sta_connect_(const WiFiAP &ap) {
         ESP_LOGV(TAG, "esp_wifi_sta_wpa2_ent_set_password failed! %d", err);
       }
     }
-    esp_wpa2_config_t wpa2_config = WPA2_CONFIG_INIT_DEFAULT();
-    err = esp_wifi_sta_wpa2_ent_enable(&wpa2_config);
+    err = esp_wifi_sta_wpa2_ent_enable();
     if (err != ESP_OK) {
       ESP_LOGV(TAG, "esp_wifi_sta_wpa2_ent_enable failed! %d", err);
     }

